units/github_status_api.go

```diff
@@ -0,0 +1,278 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/google/go-github/github"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/logging"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+const (
+	githubStatusUpdateJobName = "github-status-update"
+	githubStatusError         = "error"
+	githubStatusFailure       = "failure"
+	githubStatusPending       = "pending"
+	githubStatusSuccess       = "success"
+
+	githubUpdateTypeBuild = "build"
+	githubUpdateTypePatch = "patch"
+)
+
+func init() {
+	registry.AddJobType(githubStatusUpdateJobName, func() amboy.Job { return makeGithubStatusUpdateJob() })
+}
+
+type githubStatusUpdateJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	logger   grip.Journaler
+	env      evergreen.Environment
+
+	FetchID    string `bson:"fetch_id" json:"fetch_id" yaml:"fetch_id"`
+	UpdateType string `bson:"update_type" json:"update_type" yaml:"update_type"`
+
+	owner    string
+	repo     string
+	prNumber int
+	ref      string
+
+	urlPath     string
+	description string
+	context     string
+	ghStatus    string
+}
+
+func makeGithubStatusUpdateJob() *githubStatusUpdateJob {
```

Comment:
As an aside, should we add new degraded mode flags for github-pull request testing so we can cut off the flow of this kind of work into the system?

Reply:
done